### ISSUE TITLE
how to use tolerances in LVS?

### ISSUE THREAD

**Issue by olisnr** [IHP-GmbH/IHP-Open-PDK#203]
i have MOSFETs with 15.975µm instead of 16:
![no_match](https://github.com/user-attachments/assets/e2784fce-3d84-4951-9af5-8396ae222bdc)

what is the correct way to define tolerances for part-parameters?

**Comment by FaragElsayed2** [comment]
@olisnr Please share a small test case for this. Most probably, it's a grid issue but I need to recheck.

**Comment by olisnr** [comment]
@FaragElsayed2 yes, its a grid issue because 16µm/9 dont fit in a 5nm grid. if i change the value in the netlist, it works, but if there are many parts, that need to be changed from hand, it will be interesting to be able solve the problem via a tolerance. the testcase is: $87

[OTA33_grid.zip](https://github.com/user-attachments/files/17045252/OTA33_grid.zip)

**Comment by FaragElsayed2** [comment]
@olisnr If you're confident that the issue is grid-related, there's no need to investigate further. To resolve it, you have two options:

    - Fix the DRC grid by trying a few methods, such as adjusting the database unit/scaling.

    - Add a tolerance for the specific parameter you need for each device, similar to the approach outlined here: [KLayout LVS Global Parameters](https://www.klayout.de/doc-qt5/about/lvs_ref_global.html#h2-248). However, this would require modifying the LVS runsets, which is not generally recommended.

By default, we’ve only added tolerances for inductors due to the complexity of geometry calculations.

**Comment by FaragElsayed2** [comment]
@olisnr I believe this isn't the ideal use case, but it might be worth discussing internally first.
CC @atorkmabrains.

**Comment by MilosMitrovic411** [comment]
I think the good solution would be for "ng" to be a multiplier not a divisor. In your case, when you calculated your total width you would have to divide it yourself and that would immediately tell you that you would have to adjust your value, and plus you would be able to chose yourself do you want your finger width to be 1.775u or 1.78u.

**Comment by olisnr** [comment]
@MilosMitrovic411 would be possible, but breaks the compatibility with the commercial PDK...

**Comment by MilosMitrovic411** [comment]
@olisnr As it is now, yes, but my point is that it should be changed.

**Comment by atorkmabrains** [comment]
@sergeiandreyev and @KrzysztofHerman I believe there is nothing to do from our end. Could you please confirm?

**Comment by sergeiandreyev** [comment]
> [@sergeiandreyev](https://github.com/sergeiandreyev) and [@KrzysztofHerman](https://github.com/KrzysztofHerman) I believe there is nothing to do from our end. Could you please confirm?

yes, for now we do not plan to change this functionality in both PDKs
